From c5d95949a665c9768d369416cd2a4ecfabe54cc0 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 21 Mar 2020 10:36:19 +0000
Subject: [PATCH] DADAO-include: elf/common.h, add EM_DADAO def

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 include/elf/common.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/elf/common.h b/include/elf/common.h
index 75c4fb7e9d7..2986e9185e7 100644
--- a/include/elf/common.h
+++ b/include/elf/common.h
@@ -336,6 +336,7 @@
 #define EM_FT32         222     /* FTDI Chip FT32 high performance 32-bit RISC architecture */
 #define EM_MOXIE        223     /* Moxie processor family */
 #define EM_AMDGPU 	224 	/* AMD GPU architecture */
+#define EM_DADAO	234	/* DADAO, FIXME */
 #define EM_RISCV 	243 	/* RISC-V */
 #define EM_LANAI	244	/* Lanai 32-bit processor.  */
 #define EM_BPF		247	/* Linux BPF – in-kernel virtual machine.  */
-- 
2.17.1

