From 8914d7118d43701eec6c462852b9f31ffeca0602 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 7 Dec 2019 15:19:29 +0800
Subject: [PATCH] DADAO: gas/configure.tgt to support dadao

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 gas/configure.tgt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gas/configure.tgt b/gas/configure.tgt
index a4828c4cae..e2fe98fcf7 100644
--- a/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -58,6 +58,7 @@ case ${cpu} in
   cr16*)		cpu_type=cr16 endian=little ;;
   crisv32)		cpu_type=cris arch=crisv32 ;;
   crx*)			cpu_type=crx endian=little ;;
+  dadao*)		cpu_type=dadao endian=big ;;
   epiphany*)		cpu_type=epiphany endian=little ;;
   fido)			cpu_type=m68k ;;
   hppa*)		cpu_type=hppa ;;
@@ -182,6 +183,8 @@ case ${generic_target} in
   d30v-*-*)				fmt=elf ;;
   dlx-*-*)				fmt=elf ;;
 
+  dadao-*-*)				fmt=elf ;;
+
   bpf-*-*)				fmt=elf ;;
   epiphany-*-*)				fmt=elf ;;
 
-- 
2.17.1

