From 5629195b85b732c906069e8eecf4da0bd2f87c51 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 21 Mar 2020 11:20:22 +0000
Subject: [PATCH] DADAO: configure.ac

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 configure.ac | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/configure.ac b/configure.ac
index 7433badc217..0233d626e15 100644
--- a/configure.ac
+++ b/configure.ac
@@ -684,6 +684,10 @@ case "${target}" in
     # See <http://gcc.gnu.org/ml/gcc-patches/2004-11/msg00572.html>.
     unsupported_languages="$unsupported_languages fortran"
     ;;
+  dadao-*-*)
+    # See <http://gcc.gnu.org/ml/gcc-patches/2004-11/msg00572.html>.
+    unsupported_languages="$unsupported_languages fortran"
+    ;;
 esac
 
 # Disable libffi for some systems.
@@ -761,6 +765,9 @@ case "${target}" in
   mmix-*-*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
+  dadao-*-*)
+    noconfigdirs="$noconfigdirs target-libffi"
+    ;;
   powerpc-*-aix*)
     ;;
   rs6000-*-aix*)
@@ -1093,6 +1100,9 @@ case "${target}" in
   mmix-*-*)
     noconfigdirs="$noconfigdirs gdb"
     ;;
+  dadao-*-*)
+    noconfigdirs="$noconfigdirs gdb"
+    ;;
   mt-*-*)
     noconfigdirs="$noconfigdirs sim"
     ;;
-- 
2.17.1

