From 6a6db6a4b61a0e269af408a0bada371059ee2928 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sun, 5 Apr 2020 07:44:53 +0000
Subject: [PATCH] DADAO: .gitignore

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 4460159c842..9996e0a2a03 100644
--- a/.gitignore
+++ b/.gitignore
@@ -16,6 +16,7 @@
 *.pyc
 *.tmp
 *.a
+*.swp
 
 .deps
 .libs
@@ -30,6 +31,7 @@ libtool
 POTFILES
 *-POTFILES
 
+tags
 TAGS
 TAGS.sub
 
-- 
2.17.1

