From 2ef17708b954cc5b9be4d2eccddd63c1b81f3380 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 18 Apr 2020 06:08:45 +0000
Subject: [PATCH 3/3] DADAO-newlib-libc-include: elf.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 newlib/libc/include/elf.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/newlib/libc/include/elf.h b/newlib/libc/include/elf.h
index 206b131ea..c6c7e8bb3 100644
--- a/newlib/libc/include/elf.h
+++ b/newlib/libc/include/elf.h
@@ -343,6 +343,7 @@ typedef struct {
 #define EM_NUM		248
 
 #define EM_ALPHA	0x9026
+#define EM_DADAO	0xDADA
 
 #define EV_NONE		0
 #define EV_CURRENT	1
-- 
2.17.1

