From cbcb2997060ab8f2dff4454342363aa91eb34aea Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 6 May 2020 07:59:25 +0000
Subject: [PATCH] DADAO-fixup: newlib/libc/stdio/Makefile.in for svfwscanf.c

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 newlib/libc/stdio/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libc/stdio/Makefile.in b/newlib/libc/stdio/Makefile.in
index febcd4af8..c029fe093 100644
--- a/newlib/libc/stdio/Makefile.in
+++ b/newlib/libc/stdio/Makefile.in
@@ -1953,7 +1953,7 @@ $(lpfx)vfiwscanf.$(oext): vfwscanf.c
 	$(LIB_COMPILE) -DINTEGER_ONLY -c $(srcdir)/vfwscanf.c -o $@
 
 $(lpfx)svfwscanf.$(oext): vfwscanf.c
-	$(LIB_COMPILE) -DSTRING_ONLY -c $(srcdir)/vfwscanf.c -o $@
+	$(LIB_COMPILE) -Os -DSTRING_ONLY -c $(srcdir)/vfwscanf.c -o $@
 
 $(lpfx)svfiwscanf.$(oext): vfwscanf.c
 	$(LIB_COMPILE) -DINTEGER_ONLY -DSTRING_ONLY -c $(srcdir)/vfwscanf.c -o $@
-- 
2.17.1

