From 48d990c311e357feacbc7acc65a22fd945b9bd13 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 6 May 2020 07:19:05 +0000
Subject: [PATCH] DADAO-fixup: newlib/libc/stdio/Makefile.in for vfiprintf.o

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 newlib/libc/stdio/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libc/stdio/Makefile.in b/newlib/libc/stdio/Makefile.in
index 6cf63ac1e..febcd4af8 100644
--- a/newlib/libc/stdio/Makefile.in
+++ b/newlib/libc/stdio/Makefile.in
@@ -1904,7 +1904,7 @@ docbook: $(DOCBOOK_OUT_FILES)
 # Otherwise libtool moves vfprintf.o and subsequently can't find it.
 
 $(lpfx)vfiprintf.$(oext): vfprintf.c
-	$(LIB_COMPILE) -DINTEGER_ONLY -c $(srcdir)/vfprintf.c -o $@
+	$(LIB_COMPILE) -O1 -DINTEGER_ONLY -c $(srcdir)/vfprintf.c -o $@
 
 $(lpfx)svfprintf.$(oext): vfprintf.c
 	$(LIB_COMPILE) -DSTRING_ONLY -c $(srcdir)/vfprintf.c -o $@
-- 
2.17.1

