From d32aa5c9507258a798ba20aaa31a5fbc6e8c33d3 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 23 May 2020 13:06:24 +0000
Subject: [PATCH] DADAO-fixup: newlib/libm/math/Makefile.in for kf_rem_pio2.o

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 newlib/libm/math/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libm/math/Makefile.in b/newlib/libm/math/Makefile.in
index 0ac2b1668..5811ddf93 100644
--- a/newlib/libm/math/Makefile.in
+++ b/newlib/libm/math/Makefile.in
@@ -877,7 +877,7 @@ lib_a-w_tgamma.obj: w_tgamma.c
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -c -o lib_a-w_tgamma.obj `if test -f 'w_tgamma.c'; then $(CYGPATH_W) 'w_tgamma.c'; else $(CYGPATH_W) '$(srcdir)/w_tgamma.c'; fi`
 
 lib_a-kf_rem_pio2.o: kf_rem_pio2.c
-	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -c -o lib_a-kf_rem_pio2.o `test -f 'kf_rem_pio2.c' || echo '$(srcdir)/'`kf_rem_pio2.c
+	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -O1 -c -o lib_a-kf_rem_pio2.o `test -f 'kf_rem_pio2.c' || echo '$(srcdir)/'`kf_rem_pio2.c
 
 lib_a-kf_rem_pio2.obj: kf_rem_pio2.c
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -c -o lib_a-kf_rem_pio2.obj `if test -f 'kf_rem_pio2.c'; then $(CYGPATH_W) 'kf_rem_pio2.c'; else $(CYGPATH_W) '$(srcdir)/kf_rem_pio2.c'; fi`
-- 
2.17.1

