From 64105b7c8407cd51012d3e99ed4b97deb9525643 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 23 May 2020 13:11:08 +0000
Subject: [PATCH] DADAO-fixup: newlib/libc/stdio/Makefile.in for svfiscanf.o

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 newlib/libc/stdio/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libc/stdio/Makefile.in b/newlib/libc/stdio/Makefile.in
index 7f5d0d2bf..ee2772446 100644
--- a/newlib/libc/stdio/Makefile.in
+++ b/newlib/libc/stdio/Makefile.in
@@ -1944,7 +1944,7 @@ $(lpfx)svfscanf.$(oext): vfscanf.c
 	$(LIB_COMPILE) -DSTRING_ONLY -c $(srcdir)/vfscanf.c -o $@
 
 $(lpfx)svfiscanf.$(oext): vfscanf.c
-	$(LIB_COMPILE) -DINTEGER_ONLY -DSTRING_ONLY -c $(srcdir)/vfscanf.c -o $@
+	$(LIB_COMPILE) -O1 -DINTEGER_ONLY -DSTRING_ONLY -c $(srcdir)/vfscanf.c -o $@
 
 $(lpfx)vfwscanf.$(oext): vfwscanf.c
 	$(LIB_COMPILE) -c $(srcdir)/vfwscanf.c -o $@
-- 
2.17.1

