From badb2c1214a462d36896ba8e06c76eeb78545b48 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Fri, 29 May 2020 03:02:22 +0000
Subject: [PATCH] DADAO-fixup: newlib/libm/math/Makefile.in for ef_pow.o

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 newlib/libm/math/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libm/math/Makefile.in b/newlib/libm/math/Makefile.in
index 5811ddf93..18cc4bfbf 100644
--- a/newlib/libm/math/Makefile.in
+++ b/newlib/libm/math/Makefile.in
@@ -997,7 +997,7 @@ lib_a-ef_log10.obj: ef_log10.c
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -c -o lib_a-ef_log10.obj `if test -f 'ef_log10.c'; then $(CYGPATH_W) 'ef_log10.c'; else $(CYGPATH_W) '$(srcdir)/ef_log10.c'; fi`
 
 lib_a-ef_pow.o: ef_pow.c
-	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -c -o lib_a-ef_pow.o `test -f 'ef_pow.c' || echo '$(srcdir)/'`ef_pow.c
+	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -O1 -c -o lib_a-ef_pow.o `test -f 'ef_pow.c' || echo '$(srcdir)/'`ef_pow.c
 
 lib_a-ef_pow.obj: ef_pow.c
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_a_CFLAGS) $(CFLAGS) -c -o lib_a-ef_pow.obj `if test -f 'ef_pow.c'; then $(CYGPATH_W) 'ef_pow.c'; else $(CYGPATH_W) '$(srcdir)/ef_pow.c'; fi`
-- 
2.17.1

