From 6606140b34efbccf27a36ee1157daa2834ed911a Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Tue, 27 Feb 2024 17:18:36 +0800
Subject: [PATCH] DADAO: include/disas/dis-asm.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 include/disas/dis-asm.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/disas/dis-asm.h b/include/disas/dis-asm.h
index 2324f6b1a4..6707ec60dc 100644
--- a/include/disas/dis-asm.h
+++ b/include/disas/dis-asm.h
@@ -453,6 +453,7 @@ int print_insn_xtensa           (bfd_vma, disassemble_info*);
 int print_insn_riscv32          (bfd_vma, disassemble_info*);
 int print_insn_riscv64          (bfd_vma, disassemble_info*);
 int print_insn_riscv128         (bfd_vma, disassemble_info*);
+int print_insn_dadao            (bfd_vma, disassemble_info*);
 int print_insn_rx(bfd_vma, disassemble_info *);
 int print_insn_hexagon(bfd_vma, disassemble_info *);
 int print_insn_loongarch(bfd_vma, disassemble_info *);
-- 
2.34.1

