From 462f25db2b9181f66310800a8d1b6732e114ce5d Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Thu, 7 Dec 2023 06:32:58 +0800
Subject: [PATCH] DADAO: meson.build

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index a9c4f28247..5099a90df9 100644
--- a/meson.build
+++ b/meson.build
@@ -2828,6 +2828,7 @@ disassemblers = {
   'sparc' : ['CONFIG_SPARC_DIS'],
   'xtensa' : ['CONFIG_XTENSA_DIS'],
   'loongarch' : ['CONFIG_LOONGARCH_DIS'],
+  'dadao' : ['CONFIG_DADAO_DIS'],
 }
 
 have_ivshmem = config_host_data.get('CONFIG_EVENTFD')
-- 
2.40.1

