From c12d381dd8b47a8e6ffd83b7cc75153d0d558c77 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 28 Feb 2024 08:45:57 +0800
Subject: [PATCH] DADAO: disas/meson.build

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 disas/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/disas/meson.build b/disas/meson.build
index 4c8da01877..0ab1d61ed3 100644
--- a/disas/meson.build
+++ b/disas/meson.build
@@ -6,6 +6,7 @@ common_ss.add(when: 'CONFIG_ARM_A64_DIS', if_true: files('arm-a64.cc'))
 common_ss.add_all(when: 'CONFIG_ARM_A64_DIS', if_true: libvixl_ss)
 common_ss.add(when: 'CONFIG_ARM_DIS', if_true: files('arm.c'))
 common_ss.add(when: 'CONFIG_CRIS_DIS', if_true: files('cris.c'))
+common_ss.add(when: 'CONFIG_DADAO_DIS', if_true: files('dadao.c'))
 common_ss.add(when: 'CONFIG_HEXAGON_DIS', if_true: files('hexagon.c'))
 common_ss.add(when: 'CONFIG_HPPA_DIS', if_true: files('hppa.c'))
 common_ss.add(when: 'CONFIG_I386_DIS', if_true: files('i386.c'))
-- 
2.34.1

