From b32661c811ddcb8f1edfaf922d2a3a7be1db3958 Mon Sep 17 00:00:00 2001
From: Hao Chenqi <hchenqi@pku.edu.cn>
Date: Fri, 26 Nov 2021 16:18:41 +0800
Subject: [PATCH 2/3] DADAO-include: sysemu/arch_init.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
Signed-off-by: Hao Chenqi <hchenqi@pku.edu.cn>
---
 include/sysemu/arch_init.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/sysemu/arch_init.h b/include/sysemu/arch_init.h
index 16da279696..edfc258ebc 100644
--- a/include/sysemu/arch_init.h
+++ b/include/sysemu/arch_init.h
@@ -26,7 +26,8 @@ enum {
     QEMU_ARCH_RISCV = (1 << 19),
     QEMU_ARCH_RX = (1 << 20),
     QEMU_ARCH_AVR = (1 << 21),
-
+    QEMU_ARCH_DADAO = (1 << 22),
+    
     QEMU_ARCH_NONE = (1 << 31),
 };
 
-- 
2.17.1

