From c41b376e4a9c8ea1fbf9f11df77bdeb6821df42d Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Tue, 11 Jan 2022 22:09:20 +0800
Subject: [PATCH] DADAO: elf/elf.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 elf/elf.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/elf/elf.h b/elf/elf.h
index 2549a177d6..a0b32388b8 100644
--- a/elf/elf.h
+++ b/elf/elf.h
@@ -374,6 +374,8 @@ typedef struct
 
 #define EM_ALPHA	0x9026
 
+#define	EM_DADAO	0x0DA0
+
 /* Legal values for e_version (version).  */
 
 #define EV_NONE		0		/* Invalid ELF version */
@@ -4027,6 +4029,12 @@ enum
 #define R_NDS32_TLS_TPOFF	102
 #define R_NDS32_TLS_DESC	119
 
+/* DADAO relocations.  */
+#define R_DADAO_NONE		0	/* No reloc */
+#define R_DADAO_JUMP_SLOT	65	/* Create PLT entry */
+#define R_DADAO_COPY		66	/* Copy symbol at runtime.  */
+#define R_DADAO_RELATIVE	67	/* Adjust by program base.  */
+
 __END_DECLS
 
 #endif	/* elf.h */
-- 
2.17.1

