From 9065c4c100c51c7fd04d5db260bb85f77919a47e Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Fri, 13 Dec 2019 03:28:53 +0800
Subject: [PATCH 1001/1002] DADAO-linux-user: elfload.c

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 linux-user/elfload.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/linux-user/elfload.c b/linux-user/elfload.c
index bd43c4817d..e01ce44d2a 100644
--- a/linux-user/elfload.c
+++ b/linux-user/elfload.c
@@ -1480,6 +1480,24 @@ static void elf_core_copy_regs(target_elf_gregset_t *regs,
 
 #endif /* TARGET_XTENSA */
 
+#ifdef TARGET_DADAO
+
+#define ELF_START_MMAP 0x80000000
+#define ELF_ARCH  EM_DADAO
+
+#define ELF_CLASS ELFCLASS64
+
+static inline void init_thread(struct target_pt_regs *regs,
+                               struct image_info *infop)
+{
+    regs->pc = infop->entry;
+    regs->sp = infop->start_stack;
+}
+
+#define ELF_EXEC_PAGESIZE 4096
+
+#endif /* TARGET_DADAO */
+
 #ifndef ELF_PLATFORM
 #define ELF_PLATFORM (NULL)
 #endif
-- 
2.17.1

