From f56dd6c39d44154420e5c9c2024d3a936f66fbcd Mon Sep 17 00:00:00 2001
From: Hao Chenqi <hchenqi@pku.edu.cn>
Date: Fri, 26 Nov 2021 16:50:06 +0800
Subject: [PATCH 1001/1002] DADAO-linux-user: elfload.c

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
Signed-off-by: Hao Chenqi <hchenqi@pku.edu.cn>
---
 linux-user/elfload.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/linux-user/elfload.c b/linux-user/elfload.c
index fc9c4f12be..3cfb93acf9 100644
--- a/linux-user/elfload.c
+++ b/linux-user/elfload.c
@@ -1504,6 +1504,24 @@ static void elf_core_copy_regs(target_elf_gregset_t *regs,
 
 #endif /* TARGET_XTENSA */
 
+#ifdef TARGET_DADAO
+
+#define ELF_START_MMAP 0x80000000
+#define ELF_ARCH  EM_DADAO
+
+#define ELF_CLASS ELFCLASS64
+
+static inline void init_thread(struct target_pt_regs *regs,
+                               struct image_info *infop)
+{
+    regs->REG_PC = infop->entry;
+    regs->REG_SP = infop->start_stack;
+}
+
+#define ELF_EXEC_PAGESIZE 4096
+
+#endif /* TARGET_DADAO */
+
 #ifdef TARGET_HEXAGON
 
 #define ELF_START_MMAP 0x20000000
-- 
2.17.1

