From e7d3033bcd6094e2f2c8a026c3cb566d0e34fb23 Mon Sep 17 00:00:00 2001
From: Zhu Guangtao <zhuguangtao@pkunity.com>
Date: Sat, 15 Jan 2022 11:52:09 +0800
Subject: [PATCH] DADAO-gcc: config.gcc

Signed-off-by: Zhu Guangtao <zhuguangtao@pkunity.com>
---
 gcc/config.gcc | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index 6fcdd771d4c..0bd2293e768 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1626,6 +1626,19 @@ csky-*-*)
 		;;
 	esac
 	;;
+dadao-*-*)
+	tm_file="dbxelf.h elfos.h gnu-user.h"
+	tmake_file="${tmake_file}"
+	case ${target} in
+	dadao-*-elf*)
+	  tm_file="${tm_file} newlib-stdint.h dadao/dadao.h"
+	  ;;
+	dadao-*-gnu*)
+	  tm_file="${tm_file} glibc-stdint.h linux.h dadao/linux.h"
+	  ;;
+	esac
+        use_gcc_stdint=wrap
+        ;;
 epiphany-*-elf | epiphany-*-rtems*)
 	tm_file="${tm_file} dbxelf.h elfos.h"
 	tmake_file="${tmake_file} epiphany/t-epiphany"
-- 
2.17.1

