From 91dae7796a1b68d43e491aec82d197e09f4e578d Mon Sep 17 00:00:00 2001
From: Zhu Guangtao <zhuguangtao@pkunity.com>
Date: Sat, 15 Jan 2022 19:42:36 +0800
Subject: [PATCH] DADAO-gcc: config.gcc

Signed-off-by: Zhu Guangtao <zhuguangtao@pkunity.com>
---
 gcc/config.gcc | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index 6fcdd771d4c..9633bfafdf9 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1626,6 +1626,23 @@ csky-*-*)
 		;;
 	esac
 	;;
+dadao-*-elf)
+	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h"
+        use_gcc_stdint=wrap
+        ;;
+dadao-*-gnu | dadao-linux-gnu)
+	tm_file="${tm_file} gnu-user.h dbxelf.h elfos.h"
+        case ${target} in
+        dadao-*-gnu)
+          tm_file="${tm_file} dadao/linux.h glibc-stdint.h"
+	  tm_defines="$tm_defines DEFAULT_LIBC=LIBC_GLIBC"
+          ;;
+	dado-linux-gnu)
+	  tm_file="${tm_file} linux.h dadao/linux.h glibc-stdint.h"
+          tm_defines="$tm_defines DEFAULT_LIBC=LIBC_GLIBC"
+          ;;
+        esac
+        ;;
 epiphany-*-elf | epiphany-*-rtems*)
 	tm_file="${tm_file} dbxelf.h elfos.h"
 	tmake_file="${tmake_file} epiphany/t-epiphany"
-- 
2.17.1

