From f23e6fb744e94c9de4cd4e3d6aa596d72eaa25e9 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 8 Jan 2022 08:40:10 +0800
Subject: [PATCH] DADAO: libgcc/config.host

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 libgcc/config.host | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libgcc/config.host b/libgcc/config.host
index c529cc40f0c..08718963db2 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -595,6 +595,10 @@ csky-*-linux*)
 	extra_parts="$extra_parts crti.o crtn.o"
 	md_unwind_header=csky/linux-unwind.h
 	;;
+dadao-*-*)
+	extra_parts="crti.o crtn.o crtbegin.o crtend.o"
+	tmake_file="${tmake_file} ${cpu_type}/t-${cpu_type}"
+	;;
 epiphany-*-elf* | epiphany-*-rtems*)
 	tmake_file="$tmake_file epiphany/t-epiphany t-fdpbit epiphany/t-custom-eqsf"
 	extra_parts="$extra_parts crti.o crtint.o crtrunc.o crtm1reg-r43.o crtm1reg-r63.o crtn.o"
-- 
2.17.1

