From ba727102074c54ab27d7d61fcb9dc725a2c4f1d1 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Tue, 19 Nov 2019 09:09:53 +0800
Subject: [PATCH] DADAO: add gcc/config.gcc support

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
Signed-off-by: shiyining <1700012728@pku.edu.cn>
---
 gcc/config.gcc | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index ddd3b8f4d9d..a0313bc9bcb 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -363,6 +363,10 @@ bfin*-*)
 crisv32-*)
 	cpu_type=cris
 	;;
+dadao*)
+	cpu_type=dadao
+	target_has_targetm_common=no
+	;;
 frv*)	cpu_type=frv
 	extra_options="${extra_options} g.opt"
 	;;
@@ -1377,6 +1381,11 @@ csky-*-*)
 		;;
 	esac
 	;;
+dadao*-*-*)
+	gas=yes
+	gnu_ld=yes
+	tmake_file="${tmake_file} dadao/t-dadao"
+	;;
 epiphany-*-elf | epiphany-*-rtems*)
 	tm_file="${tm_file} dbxelf.h elfos.h"
 	tmake_file="${tmake_file} epiphany/t-epiphany"
-- 
2.17.1

