From f8daeb91f4eaeb090859e32f2c9777ea9a320ce4 Mon Sep 17 00:00:00 2001
From: Yang Chen <yang_chen@pku.edu.cn>
Date: Tue, 8 Feb 2022 19:20:42 +0800
Subject: [PATCH] DADAO-bfd: reloc.c

Signed-off-by: Yang Chen <yang_chen@pku.edu.cn>
Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 bfd/reloc.c | 29 ++++++++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/bfd/reloc.c b/bfd/reloc.c
index 9ca68ca61da..1d7c0d42ad3 100644
--- a/bfd/reloc.c
+++ b/bfd/reloc.c
@@ -8150,6 +8150,35 @@ ENUM
 ENUMDOC
   S12Z relocations.
 
+ENUM
+  BFD_RELOC_DADAO_ABS
+ENUMDOC
+  These are relocations for ABS 64bit-address.
+ENUM
+  BFD_RELOC_DADAO_BRCC
+ENUMDOC
+  These are relocations for a conditional branch instruction.
+ENUM
+  BFD_RELOC_DADAO_BRCC12
+ENUMDOC
+  These are relocations for a 12-bit conditional branch instruction.
+ENUM
+  BFD_RELOC_DADAO_CALL
+ENUMDOC
+  These are relocations for the CALL instruction.
+ENUM
+  BFD_RELOC_DADAO_JUMP
+ENUMDOC
+  These are relocations for the JUMP instruction.
+ENUM
+  BFD_RELOC_DADAO_HI18
+ENUMDOC
+  ADRP high 18-bit
+ENUM
+  BFD_RELOC_DADAO_LO12
+ENUMDOC
+  ADRP low 12-bit
+
 ENDSENUM
   BFD_RELOC_UNUSED
 CODE_FRAGMENT
-- 
2.17.1

