From 0f11a7758fde80a413baf59ac1d51bd4a873ddb0 Mon Sep 17 00:00:00 2001
From: Yang Chen <yang_chen@pku.edu.cn>
Date: Sun, 28 Nov 2021 14:04:03 +0800
Subject: [PATCH] DADAO: ld/configure.tgt

Signed-off-by: Yang Chen <yang_chen@pku.edu.cn>
---
 ld/configure.tgt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ld/configure.tgt b/ld/configure.tgt
index 87c7d9a4cad..eb31c2cf94a 100644
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -253,6 +253,8 @@ d30v-*-*)		targ_emul=d30velf
 			targ_extra_emuls="d30v_e d30v_o"
 			targ_extra_ofiles=ldelfgen.o
 			;;
+dadao-*-*)		targ_emul=elf64_dadao
+			;;
 dlx-*-elf*)		targ_emul=elf32_dlx
 			targ_extra_ofiles=ldelfgen.o
 			;;
-- 
2.17.1

