From 5dc1576ca40bd822809dc6383c5e685c22546766 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 21 Mar 2020 10:00:37 +0000
Subject: [PATCH 4002/4004] DADAO-opcodes: configure.ac

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 opcodes/configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/opcodes/configure.ac b/opcodes/configure.ac
index 500c701a62b..6421740afd8 100644
--- a/opcodes/configure.ac
+++ b/opcodes/configure.ac
@@ -267,6 +267,7 @@ if test x${all_targets} = xfalse ; then
 	bfd_csky_arch)		ta="$ta csky-dis.lo" ;;
 	bfd_d10v_arch)		ta="$ta d10v-dis.lo d10v-opc.lo" ;;
 	bfd_d30v_arch)		ta="$ta d30v-dis.lo d30v-opc.lo" ;;
+	bfd_dadao_arch)		ta="$ta dadao-dis.lo dadao-opc.lo" ;;
 	bfd_dlx_arch)		ta="$ta dlx-dis.lo" ;;
 	bfd_fr30_arch)		ta="$ta fr30-asm.lo fr30-desc.lo fr30-dis.lo fr30-ibld.lo fr30-opc.lo" using_cgen=yes ;;
 	bfd_frv_arch)		ta="$ta frv-asm.lo frv-desc.lo frv-dis.lo frv-ibld.lo frv-opc.lo" using_cgen=yes ;;
-- 
2.17.1

