From 34b65ae1518131d855b7b07fbcc0dab83551e407 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Thu, 7 May 2020 03:22:17 +0000
Subject: [PATCH] DADAO-binutils: readelf.c

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 binutils/readelf.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/binutils/readelf.c b/binutils/readelf.c
index 0962877ad8f..9614941e0c4 100644
--- a/binutils/readelf.c
+++ b/binutils/readelf.c
@@ -102,6 +102,7 @@
 #include "elf/csky.h"
 #include "elf/d10v.h"
 #include "elf/d30v.h"
+#include "elf/dadao.h"
 #include "elf/dlx.h"
 #include "elf/bpf.h"
 #include "elf/epiphany.h"
@@ -832,6 +833,7 @@ guess_is_rela (unsigned int e_machine)
     case EM_MCORE:
     case EM_CYGNUS_MEP:
     case EM_METAG:
+    case EM_DADAO:
     case EM_MMIX:
     case EM_MN10200:
     case EM_CYGNUS_MN10200:
@@ -1375,6 +1377,10 @@ dump_relocations (Filedata *          filedata,
 	  rtype = elf_mcore_reloc_type (type);
 	  break;
 
+	case EM_DADAO:
+	  rtype = elf_dadao_reloc_type (type);
+	  break;
+
 	case EM_MMIX:
 	  rtype = elf_mmix_reloc_type (type);
 	  break;
@@ -2542,6 +2548,7 @@ get_machine_name (unsigned e_machine)
     case EM_CYGNUS_FRV:		return "Fujitsu FR-V";
     case EM_S12Z:               return "Freescale S12Z";
     case EM_CSKY:		return "C-SKY";
+    case EM_DADAO: 	 	return "DADAO 64-bit processor";
 
     default:
       snprintf (buff, sizeof (buff), _("<unknown>: 0x%x"), e_machine);
@@ -12634,6 +12641,8 @@ is_32bit_abs_reloc (Filedata * filedata, unsigned int reloc_type)
       return reloc_type == 1; /* R_MICROBLAZE_32.  */
     case EM_MIPS:
       return reloc_type == 2; /* R_MIPS_32.  */
+    case EM_DADAO:
+      return reloc_type == 4; /* R_DADAO_32.  */
     case EM_MMIX:
       return reloc_type == 4; /* R_MMIX_32.  */
     case EM_CYGNUS_MN10200:
@@ -12850,6 +12859,8 @@ is_64bit_abs_reloc (Filedata * filedata, unsigned int reloc_type)
       return reloc_type == 1; /* R_TILEGX_64.  */
     case EM_MIPS:
       return reloc_type == 18;	/* R_MIPS_64.  */
+    case EM_DADAO:
+      return reloc_type == 5;  /* R_DADAO_64.  */
     default:
       return FALSE;
     }
-- 
2.17.1

