From 387308f4f9f8423f55c77fef1d31f1917bcb2133 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 11 Apr 2020 12:08:56 +0000
Subject: [PATCH] DADAO-libgcc: config.host

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 libgcc/config.host | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libgcc/config.host b/libgcc/config.host
index 0f15fda3612..7a527231223 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -530,6 +530,10 @@ csky-*-linux*)
 	extra_parts="$extra_parts crti.o crtn.o"
 	md_unwind_header=csky/linux-unwind.h
 	;;
+dadao-*-*)
+	extra_parts="crti.o crtn.o crtbegin.o crtend.o"
+	tmake_file="${tmake_file} ${cpu_type}/t-${cpu_type}"
+	;;
 epiphany-*-elf* | epiphany-*-rtems*)
 	tmake_file="$tmake_file epiphany/t-epiphany t-fdpbit epiphany/t-custom-eqsf"
 	extra_parts="$extra_parts crti.o crtint.o crtrunc.o crtm1reg-r43.o crtm1reg-r63.o crtn.o"
-- 
2.17.1

