From f85e1ca4c7ed6dd4119796f9ddd49d738778a42a Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 8 Apr 2020 11:49:44 +0000
Subject: [PATCH] DADAO: config.sub

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 config.sub | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config.sub b/config.sub
index 75bb6a3135b..0e92f211d37 100755
--- a/config.sub
+++ b/config.sub
@@ -1174,6 +1174,7 @@ case $cpu-$vendor in
 			| c[123]* | c30 | [cjt]90 | c4x \
 			| c8051 | clipper | craynv | csky | cydra \
 			| d10v | d30v | dlx | dsp16xx \
+			| dadao \
 			| e2k | elxsi | epiphany \
 			| f30[01] | f700 | fido | fr30 | frv | ft32 | fx80 \
 			| h8300 | h8500 \
-- 
2.17.1

