From 6a12b23291efb1008542f8359fed4285a04714e2 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 11 Dec 2019 14:13:41 +0800
Subject: [PATCH] DADAO: configure add dadao support

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index 714e7fb6a1..a3d58c7a85 100755
--- a/configure
+++ b/configure
@@ -7513,6 +7513,9 @@ case "$target_name" in
   cris)
     target_compiler=$cross_cc_cris
   ;;
+  dadao)
+    target_compiler=$cross_cc_dadao
+  ;;
   hppa)
     mttcg="yes"
     target_compiler=$cross_cc_hppa
-- 
2.17.1

