From eadd7f734d8a0a3402fb636ad85caa551dccf05c Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Fri, 13 Dec 2019 03:29:45 +0800
Subject: [PATCH 3/3] DADAO: linux-user/syscall_defs.h add TARGET_DADAO support

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 linux-user/syscall_defs.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/linux-user/syscall_defs.h b/linux-user/syscall_defs.h
index 0662270300..8a07dd536f 100644
--- a/linux-user/syscall_defs.h
+++ b/linux-user/syscall_defs.h
@@ -73,7 +73,7 @@
     || defined(TARGET_S390X) \
     || defined(TARGET_OPENRISC) || defined(TARGET_TILEGX) \
     || defined(TARGET_NIOS2) || defined(TARGET_RISCV) \
-    || defined(TARGET_XTENSA)
+    || defined(TARGET_XTENSA) || defined(TARGET_DADAO)
 
 #define TARGET_IOC_SIZEBITS	14
 #define TARGET_IOC_DIRBITS	2
@@ -1952,7 +1952,7 @@ struct target_stat64  {
 };
 
 #elif defined(TARGET_OPENRISC) || defined(TARGET_TILEGX) || \
-      defined(TARGET_NIOS2) || defined(TARGET_RISCV)
+      defined(TARGET_NIOS2) || defined(TARGET_RISCV) || defined(TARGET_DADAO)
 
 /* These are the asm-generic versions of the stat and stat64 structures */
 
@@ -2125,7 +2125,7 @@ struct target_statfs64 {
 };
 #elif (defined(TARGET_PPC64) || defined(TARGET_X86_64) || \
        defined(TARGET_SPARC64) || defined(TARGET_AARCH64) || \
-       defined(TARGET_RISCV)) && !defined(TARGET_ABI32)
+       defined(TARGET_RISCV)) && !defined(TARGET_ABI32) || defined (TARGET_DADAO)
 struct target_statfs {
 	abi_long f_type;
 	abi_long f_bsize;
-- 
2.17.1

