From c952879d78bd77686e4fddc5026a1786f51eed63 Mon Sep 17 00:00:00 2001
From: Guorui Wen <grwen@stu.pku.edu.cn>
Date: Wed, 22 May 2024 12:11:27 +0800
Subject: [PATCH 10/11] DADAO: llvm/lib/CodeGen/ValueTypes.cpp

Signed-off-by: Guorui Wen <grwen@stu.pku.edu.cn>
---
 llvm/lib/CodeGen/ValueTypes.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/llvm/lib/CodeGen/ValueTypes.cpp b/llvm/lib/CodeGen/ValueTypes.cpp
index 608434800bc3..ba8e1e69555d 100644
--- a/llvm/lib/CodeGen/ValueTypes.cpp
+++ b/llvm/lib/CodeGen/ValueTypes.cpp
@@ -173,6 +173,7 @@ std::string EVT::getEVTString() const {
   case MVT::Untyped:   return "Untyped";
   case MVT::funcref:   return "funcref";
   case MVT::externref: return "externref";
+  case MVT::bp64:  return "bp64";
   }
 }
 
@@ -207,6 +208,8 @@ Type *EVT::getTypeForEVT(LLVMContext &Context) const {
   case MVT::externref:
     // pointer to opaque struct in addrspace(10)
     return PointerType::get(StructType::create(Context), 10);
+  case MVT::bp64:
+    return PointerType::get(Context, 0);
   case MVT::funcref:
     // pointer to i8 addrspace(20)
     return PointerType::get(Type::getInt8Ty(Context), 20);
-- 
2.34.1

