From ed33e2922274da3d96c3c273404c33b9ea78982d Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Thu, 11 May 2023 10:58:18 +0800
Subject: [PATCH] DADAO: llvm/include/llvm/BinaryFormat/ELF.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 llvm/include/llvm/BinaryFormat/ELF.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/llvm/include/llvm/BinaryFormat/ELF.h b/llvm/include/llvm/BinaryFormat/ELF.h
index 75f0c960beea..00a891c3ad1b 100644
--- a/llvm/include/llvm/BinaryFormat/ELF.h
+++ b/llvm/include/llvm/BinaryFormat/ELF.h
@@ -316,6 +316,7 @@ enum {
   EM_AMDGPU = 224,        // AMD GPU architecture
   EM_RISCV = 243,         // RISC-V
   EM_LANAI = 244,         // Lanai 32-bit processor
+  EM_DADAO = 0x0DA0,      // Dadao 64-bit processor
   EM_BPF = 247,           // Linux kernel bpf virtual machine
   EM_VE = 251,            // NEC SX-Aurora VE
   EM_CSKY = 252,          // C-SKY 32-bit processor
@@ -665,6 +666,11 @@ enum {
 #include "ELFRelocs/Lanai.def"
 };
 
+// ELF Relocation type for Dadao.
+enum {
+#include "ELFRelocs/Dadao.def"
+};
+
 // RISCV Specific e_flags
 enum : unsigned {
   EF_RISCV_RVC = 0x0001,
-- 
2.34.1

