From 6f23df6f684a3e06a64c2615024f9dfb1dffbc67 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Thu, 11 May 2023 11:01:08 +0800
Subject: [PATCH] DADAO: llvm/lib/BinaryFormat/ELF.cpp

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 llvm/lib/BinaryFormat/ELF.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/llvm/lib/BinaryFormat/ELF.cpp b/llvm/lib/BinaryFormat/ELF.cpp
index dc8f3051a149..f3e9a9ba5113 100644
--- a/llvm/lib/BinaryFormat/ELF.cpp
+++ b/llvm/lib/BinaryFormat/ELF.cpp
@@ -194,6 +194,7 @@ uint16_t ELF::convertArchNameToEMachine(StringRef Arch) {
       .Case("amdgpu", EM_AMDGPU)
       .Case("riscv", EM_RISCV)
       .Case("lanai", EM_LANAI)
+      .Case("dadao", EM_DADAO)
       .Case("bpf", EM_BPF)
       .Case("ve", EM_VE)
       .Case("csky", EM_CSKY)
@@ -556,6 +557,8 @@ StringRef ELF::convertEMachineToArchName(uint16_t EMachine) {
     return "riscv";
   case EM_LANAI:
     return "lanai";
+  case EM_DADAO:
+    return "dadao";
   case EM_BPF:
     return "bpf";
   case EM_VE:
-- 
2.34.1

