From 4f004885784d4d301b7332451c924838f80d3561 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sat, 18 Apr 2020 06:01:58 +0000
Subject: [PATCH] DADAO-include: elf/common.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 include/elf/common.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/elf/common.h b/include/elf/common.h
index 75c4fb7e9d7..3201eda1b09 100644
--- a/include/elf/common.h
+++ b/include/elf/common.h
@@ -415,6 +415,9 @@
 #define EM_CYGNUS_MN10300	0xbeef
 #define EM_CYGNUS_MN10200	0xdead
 
+/* DADAO */
+#define EM_DADAO		0xDADA
+
 /* Renesas M32C and M16C.  */
 #define EM_M32C_OLD		0xFEB0
 
-- 
2.17.1

