From 30a1ff8fa6fab3378df593cf40fcd3d8ad5fb389 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Mon, 6 Apr 2020 09:09:54 +0000
Subject: [PATCH] DADAO-bfd: reloc.c

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 bfd/reloc.c | 40 +++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/bfd/reloc.c b/bfd/reloc.c
index 9ca68ca61da..1d7c0d42ad3 100644
--- a/bfd/reloc.c
+++ b/bfd/reloc.c
@@ -8150,6 +8150,46 @@ ENUM
 ENUMDOC
   S12Z relocations.
 
+ENUM
+  BFD_RELOC_DADAO_GETA
+ENUMDOC
+  These are relocations for the GETA instruction.
+ENUM
+  BFD_RELOC_DADAO_CBRANCH
+ENUMDOC
+  These are relocations for a conditional branch instruction.
+ENUM
+  BFD_RELOC_DADAO_CALL
+ENUMDOC
+  These are relocations for the CALL instruction.
+ENUM
+  BFD_RELOC_DADAO_JMP
+ENUMDOC
+  These are relocations for the JMP instruction.
+ENUM
+  BFD_RELOC_DADAO_LDST
+ENUMDOC
+  These are relocations for the LD/ST instruction.
+ENUM
+  BFD_RELOC_DADAO_ADDR19
+ENUMDOC
+  This is a relocation for a relative address as in a GETA instruction or
+  a branch.
+ENUM
+  BFD_RELOC_DADAO_ADDR27
+ENUMDOC
+  This is a relocation for a relative address as in a JMP instruction.
+ENUM
+  BFD_RELOC_DADAO_REG
+ENUMDOC
+  This is a relocation for an instruction field that may be a general
+  register.
+ENUM
+  BFD_RELOC_DADAO_BASE_PLUS_OFFSET
+ENUMDOC
+  This is a relocation for two instruction fields holding a register and
+  an offset, the equivalent of the relocation.
+
 ENDSENUM
   BFD_RELOC_UNUSED
 CODE_FRAGMENT
-- 
2.17.1

