From ff18c0af61f9a59f86aad1823034db3981d71b77 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Tue, 28 Apr 2020 09:45:15 +0000
Subject: [PATCH] DADAO-gas: testsuite/gas/all/gas.exp

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 gas/testsuite/gas/all/gas.exp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gas/testsuite/gas/all/gas.exp b/gas/testsuite/gas/all/gas.exp
index 4f0e98029cf..d109ecba379 100644
--- a/gas/testsuite/gas/all/gas.exp
+++ b/gas/testsuite/gas/all/gas.exp
@@ -112,6 +112,7 @@ if { ![is_aout_format] } {
 	}
 	{ alpha-*-*linux*ecoff } { }
 	{ alpha-*-osf* } { }
+	{ dadao-*-* } { }
 	{ hppa*-*-hpux* } { }
 	{ mep-*-* } { }
 	{ mmix-*-* } { }
@@ -282,6 +283,7 @@ case $target_triplet in {
 
 # '<' and '>' appear to have special meanings on the excluded targets
 case $target_triplet in {
+    { dadao-*-* } { }
     { frv-*-* } { }
     { hppa*-*-* } { }
     { m32r-*-* } { }
@@ -430,6 +432,9 @@ case $target_triplet in {
     { "bpf-*-*" } {
         set nop_type 6
     }
+    { "dadao-*-*" } {
+	set nop_type 5
+    }
     { "mmix-*-*" } {
 	set nop_type 5
     }
-- 
2.17.1

