From c7001a70bcb0c6d851ba09aed1ea9fbfc74be9e8 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 12 Jun 2024 15:35:47 +0800
Subject: [PATCH] DADAO: common.mk

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 common.mk | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/common.mk b/common.mk
index 462f1945..2b3a42f0 100644
--- a/common.mk
+++ b/common.mk
@@ -115,7 +115,7 @@ endif
 $(build_dir):
 	mkdir -p $@
 
-$(BOOTROM_TARGETS): $(build_dir)/bootrom.%.img: $(TESTCHIP_RSRCS_DIR)/testchipip/bootrom/bootrom.%.img | $(build_dir)
+$(BOOTROM_TARGETS): $(build_dir)/bootrom.%.img: $(BOOTROM_FILES_DIR)/bootrom.%.img | $(build_dir)
 	cp -f $< $@
 
 #########################################################################################
@@ -369,7 +369,8 @@ run-binary: check-binary $(BINARY).run
 run-binaries: check-binaries $(addsuffix .run,$(BINARIES))
 
 %.run: %.check-exists $(SIM_PREREQ) | $(output_dir)
-	(set -o pipefail && $(NUMA_PREFIX) $(sim) $(PERMISSIVE_ON) $(call get_common_sim_flags,$*) $(VERBOSE_FLAGS) $(PERMISSIVE_OFF) $* </dev/null 2> >(spike-dasm > $(call get_sim_out_name,$*).out) | tee $(call get_sim_out_name,$*).log)
+	(set -o pipefail && $(NUMA_PREFIX) $(sim) $(PERMISSIVE_ON) $(call get_common_sim_flags,$*) $(VERBOSE_FLAGS) $(PERMISSIVE_OFF) $* </dev/null 2> >($(RISCV)/__opcodes/parse-opcodes.py --input $(RISCV)/__opcodes/simrisc.opcodes --dasm dadao > $(call get_sim_out_name,$*).out) | tee $(call get_sim_out_name,$*).log)
+
 
 # run simulator as fast as possible (no insn disassembly)
 run-binary-fast: check-binary $(BINARY).run.fast
-- 
2.34.1

