From fd45622ea55775702d059ca772e207b446111d40 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Wed, 18 Jan 2023 10:40:44 +0800
Subject: [PATCH] DADAO: common.mk

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 common.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common.mk b/common.mk
index 1993c475..0872005b 100644
--- a/common.mk
+++ b/common.mk
@@ -95,7 +95,7 @@ endif
 $(build_dir):
 	mkdir -p $@
 
-$(BOOTROM_TARGETS): $(build_dir)/bootrom.%.img: $(TESTCHIP_RSRCS_DIR)/testchipip/bootrom/bootrom.%.img | $(build_dir)
+$(BOOTROM_TARGETS): $(build_dir)/bootrom.%.img: $(BOOTROM_FILES_DIR)/bootrom.%.img | $(build_dir)
 	cp -f $< $@
 
 #########################################################################################
@@ -203,7 +203,7 @@ endif
 
 # run normal binary with hardware-logged insn dissassembly
 run-binary: $(output_dir) $(SIM_PREREQ) check-binary
-	(set -o pipefail && $(NUMA_PREFIX) $(sim) $(PERMISSIVE_ON) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(SEED_FLAG) $(VERBOSE_FLAGS) $(PERMISSIVE_OFF) $(BINARY) </dev/null 2> >(spike-dasm > $(sim_out_name).out) | tee $(sim_out_name).log)
+	(set -o pipefail && $(NUMA_PREFIX) $(sim) $(PERMISSIVE_ON) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(SEED_FLAG) $(VERBOSE_FLAGS) $(PERMISSIVE_OFF) $(BINARY) </dev/null 2> >($(RISCV)/__opcodes/parse-opcodes.py --input $(RISCV)/__opcodes/simrisc.opcodes --dasm dadao > $(sim_out_name).out) | tee $(sim_out_name).log)
 
 # run simulator as fast as possible (no insn disassembly)
 run-binary-fast: $(output_dir) $(SIM_PREREQ) check-binary
-- 
2.34.1

