From d81812fef2797f12b5382a898526ce83b607888a Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@pku.edu.cn>
Date: Sun, 18 Dec 2022 18:33:53 +0800
Subject: [PATCH] DADAO-rte: fesvr/elf.h

Signed-off-by: Guan Xuetao <gxt@pku.edu.cn>
---
 fesvr/elf.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fesvr/elf.h b/fesvr/elf.h
index 7b38bf11..b2f09f75 100644
--- a/fesvr/elf.h
+++ b/fesvr/elf.h
@@ -7,6 +7,7 @@
 
 #define ET_EXEC 2
 #define EM_RISCV 243
+#define EM_DADAO 0x0DA0
 #define EM_NONE 0
 #define EV_CURRENT 1
 
@@ -22,6 +23,7 @@
 #define IS_ELFBE(hdr) (IS_ELF(hdr) && (hdr).e_ident[5] == 2)
 #define IS_ELF_EXEC(hdr) (IS_ELF(hdr) && ELF_SWAP((hdr), (hdr).e_type) == ET_EXEC)
 #define IS_ELF_RISCV(hdr) (IS_ELF(hdr) && ELF_SWAP((hdr), (hdr).e_machine) == EM_RISCV)
+#define IS_ELF_DADAO(hdr) (IS_ELF(hdr) && ELF_SWAP((hdr), (hdr).e_machine) == EM_DADAO)
 #define IS_ELF_EM_NONE(hdr) (IS_ELF(hdr) && ELF_SWAP((hdr), (hdr).e_machine) == EM_NONE)
 #define IS_ELF_VCURRENT(hdr) (IS_ELF(hdr) && ELF_SWAP((hdr), (hdr).e_version) == EV_CURRENT)
 
-- 
2.34.1

